### PR TITLE
PLGMAGTWOS-484: Rest Api to retrieve payment url

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by mikededecker1** [comment]
Hi @dheesbeen 

Thank you for you pull request.
We will look into the changes and will let you know if any changes are requested.

Keep you updated.

File: Model/GuestCart/PaymentUrl.php
Comment (mikededecker1) [inline Model/GuestCart/PaymentUrl.php:41]: It is possible to use an invalid Quote_Masked_Id. If this is being used, you can recieve the payment link with just the order id. because of privacy reasons we can't merge this Pull request until the payment link can be recieved with the Quote_Masked Id and order Id. 

This can be confirmed by reaching the URL:
```https://example.com/rest/V1/multisafepay/invalid_value/payment-url/{{orderId}}```

I think the solution is to throw an Exception if ```$quoteIdMask->getQuoteId() === null```

**Review by mikededecker1 — CHANGES_REQUESTED** [review]
Hi @dheesbeen 

After reviewing your pull request I have found an issue regarding the getPaymentUrl function for the Guest. 
This problem should be resolved before merging this pull request. 

If you allow it, I can add a commit to solve this problem.

File: Model/GuestCart/PaymentUrl.php
Comment (ghost) [reply Model/GuestCart/PaymentUrl.php:41]: Nice find. We are testing this api today in our PWA. I will update the pull request

**Pull request merged**
